Changed Owncloud to Nextcloud

arpitbatra123 opened the pull request:
Changed OwnCloud to Nextcloud. Please merge if everything works.
Happy Hacktoberfest.

arpitbatra123 committed:
Add NextCloud Config

arpitbatra123 committed:
Update Dockerfile. Changed To NextCloud

arpitbatra123 committed:
Delete 001-owncloud.conf

brejoc commented:
Did you try to build the image? Looks like `libapache2-mod-php7.0` cannot be found:

```
Sending build context to Docker daemon  96.26kB
Step 1/25 : FROM phusion/baseimage:0.9.15
 ---> 4cb79d39875e
Step 2/25 : MAINTAINER Jochen Breuer "brejoc@gmail.com"
 ---> Using cache
 ---> d69b79b22ba8
Step 3/25 : RUN apt-get -y update
 ---> Using cache
 ---> 1e8bedd281dd
Step 4/25 : RUN apt-get install -y apache2 mariadb-server libapache2-mod-php7.0
 ---> Running in 0d5069f90018
Reading package lists...
Building dependency tree...
Reading state information...
E: Unable to locate package libapache2-mod-php7.0
E: Couldn't find any package by regex 'libapache2-mod-php7.0'
The command '/bin/sh -c apt-get install -y apache2 mariadb-server libapache2-mod-php7.0' returned a non-zero code: 100
```

arpitbatra123 committed:
Modified Dockerfile

arpitbatra123 commented:
made the required changes in #16 Please Check.

brejoc commented:
I've noticed several issues:

1. There is no driver for SQLite and MySQL installed. Both should be available.
2. PHP modules are missing:
    * zip
    * dom
    * XMLWriter
    * XMLReader
    * libxml
    * mb multibyte
    * GD
    * SimpleXML
    * cURL
3. You've configured Apache to map Nextcloud to the folder `/nextcloud`. This is unnecessary and Should be avoided.

brejoc commented on Dockerfile:
```diff
@@ -1,24 +1,34 @@
-FROM phusion/baseimage:0.9.15
-MAINTAINER Jochen Breuer "brejoc@gmail.com"
+FROM phusion/baseimage:latest
+MAINTAINER Jochen Breuer "arpitbatra123@gmail.com"
```
You shouldn't change that for my repo. ;)

brejoc commented on Dockerfile:
```diff
@@ -1,24 +1,34 @@
+MAINTAINER Jochen Breuer "arpitbatra123@gmail.com"
 RUN apt-get -y update
-RUN apt-get install -y apache2 php5 php5-gd php-xml-parser php5-intl php5-mysqlnd php5-json php5-mcrypt php5-sqlite smbclient curl libcurl3 php5-curl bzip2 wget
+RUN add-apt-repository -y ppa:ondrej/php
```
Do we really need a PPA here?

arpitbatra123 replied:
I couldn't get it to work without the PPA.

brejoc replied:
Okay, then I'm fine with it. The PPA seems to be well maintained.

brejoc commented on Dockerfile:
```diff
@@ -1,24 +1,34 @@
+RUN apt-get update  
+RUN apt-get install -y --force-yes apache2 mariadb-server libapache2-mod-php
+RUN apt-get install -y  --force-yes php7.0-gd php7.0-json php7.0-mysql php7.0-curl php7.0-mbstring
+RUN apt-get install -y  --force-yes php7.0-intl php7.0-mcrypt php-imagick php7.0-xml php7.0-zip
```
Like mentioned in my last comment, there are some PHP modules missing.

arpitbatra123 replied:
Added the required Php modules.

brejoc replied:
Sorry, I just also noticed, that SQLite is missing. It should be possible to use this with MariaDB/MySQL *and* SQLite.

arpitbatra123 replied:
Ok , I will add the sqlite3 package

arpitbatra123 replied:
Added the sqlite package.

brejoc commented on Dockerfile:
```diff
@@ -1,24 +1,34 @@
-FROM phusion/baseimage:0.9.15
-MAINTAINER Jochen Breuer "brejoc@gmail.com"
+FROM phusion/baseimage:latest
```
Please pin this to a specific version. Otherwise this will silently break in the future.

arpitbatra123 replied:
OK.

brejoc commented on Dockerfile:
```diff
@@ -1,24 +1,34 @@
+RUN apt-get install -y --force-yes apache2 mariadb-server libapache2-mod-php
+RUN apt-get install -y  --force-yes php7.0-gd php7.0-json php7.0-mysql php7.0-curl php7.0-mbstring
+RUN apt-get install -y  --force-yes php7.0-intl php7.0-mcrypt php-imagick php7.0-xml php7.0-zip
+RUN apt-get install -y wget unzip
```
We don't need `wget` and `unzip`. See below.

brejoc commented on Dockerfile:
```diff
@@ -1,24 +1,34 @@
-RUN mkdir /var/www/owncloud/data
-RUN chown -R www-data:www-data /var/www/owncloud
-RUN chmod 770 -R /var/www/owncloud/data
+RUN wget https://download.nextcloud.com/server/releases/nextcloud-12.0.3.zip
```
Doing something like this would be more elegant: `curl -k https://download.nextcloud.com/server/releases/nextcloud-12.0.3.tar.bz2 | tar jx -C /var/www/`

This would download and expand the archive on the fly. Also bzip2 is more efficient that zip.

arpitbatra123 replied:
OK

brejoc commented on Dockerfile:
```diff
@@ -1,24 +1,34 @@
-RUN ln -s /etc/apache2/sites-available/001-owncloud.conf /etc/apache2/sites-enabled/
+RUN ln -s /etc/apache2/sites-available/nextcloud.conf /etc/apache2/sites-enabled/nextcloud.conf
 RUN a2enmod rewrite
+RUN a2enmod headers
```
Why do we need `header`, `env`, `dir` and `mime`?

arpitbatra123 replied:
These modules are recommended by nextcloud install docs.

brejoc commented on nextcloud.conf:
```diff
@@ -0,0 +1,14 @@
+Alias /nextcloud "/var/www/nextcloud/"
```
Please don't map this to a subfolder. Nextcloud is the only app used here.

Commit:
Fixed Php Modules

Commit:
Fixed Config file

arpitbatra123 commented:
I made some commits. Can you please check again?

brejoc commented on Dockerfile:
```diff
@@ -1,18 +1,17 @@
 
-RUN wget https://download.nextcloud.com/server/releases/nextcloud-12.0.3.zip
-RUN unzip nextcloud-12.0.3.zip -d /var/www 
+RUN curl -k https://download.nextcloud.com/server/releases/nextcloud-12.0.3.tar.bz2 | tar jx -C /var/www/ 
```
This is not working, because we are missing the package `bzip2` to actually unpack the archive. You should add that somewhere above.

brejoc reviewed (changes requested):
Besides the missing `bzip2` package, the apache-config is not working. Perhaps you should switch back to the old config and just update the paths.

arpitbatra123 commented:
I'm on it.

arpitbatra123 committed:
Added missing bz2 package and reverted to old apache config, swapped out paths.

arpitbatra123 commented:
I did the requested changes. Can you please check again?

arpitbatra123 commented:
I did the requested changes. Can you please check again?

brejoc commented:
Seems like SQLite still isn't working. I'd guess that the php bindings for SQLite are missing.

arpitbatra123 committed:
Updated DockerFile - Added Sqlite3 as Database Choice
Added Sqlite3 and it's Corresponding php module,
Sqlite works out of the box on nextcloud installation and it has been tested.
Mysql however requires creating a new user which has not been implemented yet.

arpitbatra123 commented:
Added the sqlite3 package and its corresponding php package.SQLite works under testing.Please Check.

brejoc commented:
@arpitbatra123 Okay, I've tested this again yesterday and there is only one small change left. Please use update to the most current version of Nextcloud, otherwise this won't work because of the new PHP version we've got there: `https://download.nextcloud.com/server/releases/nextcloud-13.0.1.tar.bz2`

Thanks!

brejoc commented:
@arpitbatra123 Thanks for your PR and for sticking with it for so long.

arpitbatra123 commented:
The pleasure was all mine.
